evaporate.js

```diff
@@ -1847,6 +1777,55 @@
         return code.length ? {code: code, msg: msg} : {};
     }
 
+    var Promise = function () {
```

Comment:
There is a native [Promise](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Promise) support in [most of modern browsers](http://caniuse.com/#feat=promises), why not use that?